### ISSUE TITLE
ContentResolver.query selection and selectionArgs parameters ignored

### ISSUE THREAD

**Issue by neobenedict** [xamarin/xamarin-android#5788]
### Steps to Reproduce

1. Perform a contentresolver query on a directory tree like the code below
2. Observe the parameters being ignored (no filtering is taking place). The parameters of selection and selectionArgs don't really matter, put garbage in that should cause a crash.

```cs
var ctx = Android.App.Application.Context;
var AppContentResolver = ctx.ContentResolver;

var children = new Uri("content://com.android.externalstorage.documents/tree/primary%3AAndroid/document/primary%3ADocuments/children");

string[] projection = {
    DocumentsContract.Document.ColumnDocumentId,
    DocumentsContract.Document.ColumnLastModified
};

var fileSelectionQuery = "document_id IN (?)";
var fileSelection = new string[] {"Test"};

var c = AppContentResolver.Query(children, projection, fileSelectionQuery, fileSelection, null);

while (c.MoveToNext())
{
    var fPath = c.GetString(0);
    var lastModified = c.GetString(1);
    // do stuff
}
```

### Expected Behavior
The selection parameter is treated as part of the query, and the results are filtered

### Actual Behavior
selection and selectionArgs are ignored completely. Complete garbage can be put into these parameters and no error is thrown.

It is rumored this is an AndroidX api bug but I can't find much info.

### Version Information

<details>

Microsoft Visual Studio Enterprise 2019
Version 16.8.2
VisualStudio.16.Release/16.8.2+30717.126
Microsoft .NET Framework
Version 4.8.03752

Installed Version: Enterprise

Visual C++ 2019   00433-90000-04738-AA931
Microsoft Visual C++ 2019

.NET Portability Analyzer   1.1.10808.0
Evaluates portability of assemblies across .NET platforms.

ASP.NET and Web Tools 2019   16.8.553.28003
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2052803+84e121f1403378489b842e1797df2f3f5a49ac3c
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.8.553.28003
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.8.553.28003
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.8.553.28003
Azure Functions and Web Jobs Tools

Bundler & Minifier   2.8.396
Adds support for bundling and minifying JavaScript, CSS and HTML files in any project.

C# Tools   3.8.0-5.20567.16+53c5d7d3cf13d88978744a32a27c5f8350a8400a
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

JetBrains ReSharper Ultimate 2020.1   Build 201.0.20200415.3103
JetBrains ReSharper Ultimate package for Microsoft Visual Studio. For more information about ReSharper Ultimate, visit http://www.jetbrains.com/resharper. Copyright © 2021 JetBrains, Inc.

JustDecompilePackage2017 Extension   1.0
JustDecompilePackage2017 Visual Studio Extension Detailed Info

Markdown Editor   1.12.253
A full featured Markdown editor with live preview and syntax highlighting. Supports GitHub flavored Markdown.

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.30924.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.8.43 (00471f8)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.8.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62010.06180
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   16.0.21016.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.8.0-5.20567.16+53c5d7d3cf13d88978744a32a27c5f8350a8400a
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual C++ for Cross Platform Mobile Development (Android)   16.0.30608.117
Visual C++ for Cross Platform Mobile Development (Android)

Visual F# Tools   16.8.0-beta.20507.4+da6be68280c89131cdba2045525b80890401defd
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.8.000.255 (d16-8@d002176)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.8.0.507 (remotes/origin/d16-8@e87b24884)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.8.112 (86385a3)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   11.1.0.17 (d16-8/c0e2b8e)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: be2226b
    Java.Interop: xamarin/java.interop/d16-8@79d9533
    ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
    SQLite: xamarin/sqlite/3.32.1@1a3276b
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc

Xamarin.iOS and Xamarin.Mac SDK   14.4.1.3 (e30c41de3)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

</details>

**Comment by neobenedict** [comment]
If I try to use the other overload with a bundle I get
```
Java.Lang.NoSuchMethodError: no non-static method "Landroid/content/ContentResolver;.query(Landroid/net/Uri;[Ljava/lang/String;Landroid/os/Bundle;Landroid/os/CancellationSignal;)Landroid/database/Cursor;"
  at Java.Interop.JniEnvironment+InstanceMethods.GetMethodID (Java.Interop.JniObjectReference type, System.String name, System.String signature) [0x0005b] in <89755ea61d9c4ae0a40ce90b872c9e2d>:0 
  at Java.Interop.JniType.GetInstanceMethod (System.String name, System.String signature) [0x0000c] in <89755ea61d9c4ae0a40ce90b872c9e2d>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.GetMethodInfo (System.String encodedMember) [0x00031] in <89755ea61d9c4ae0a40ce90b872c9e2d>:0 
  at Java.Interop.JniPeerMembers+JniInstanceMethods.InvokeNonvirtualObjectMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters) [0x00006] in <89755ea61d9c4ae0a40ce90b872c9e2d>:0 
  at Android.Content.ContentResolver.Query (Android.Net.Uri uri, System.String[] projection, Android.OS.Bundle queryArgs, Android.OS.CancellationSignal cancellationSignal) [0x00097] in <a54fd02a94254b8f91efc9ea3cec736c>:0 
  at RayshiftTranslateFGO.Droid.ContentManager.GetFolderChildren (Android.Net.Uri uri, System.String path, System.String[] fileSelection) [0x000c4] in G:\Projects\Rayshift\RayshiftTranslateFGO\RayshiftTranslateFGO.Android\Services\ContentManager.cs:411 
  --- End of managed Java.Lang.NoSuchMethodError stack trace ---
java.lang.NoSuchMethodError: no non-static method "Landroid/content/ContentResolver;.query(Landroid/net/Uri;[Ljava/lang/String;Landroid/os/Bundle;Landroid/os/CancellationSignal;)Landroid/database/Cursor;"
	at mono.java.lang.RunnableImplementor.n_run(Native Method)
	at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5293)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

```cs
Bundle? queryArgs = new Bundle();
if (fileSelection != null)
{
    var sb = new StringBuilder();
    foreach (var _ in fileSelection)
    {
        sb.Append("?,");
    }

    var inQuery = sb.ToString().TrimEnd(',');
    fileSelectionQuery = DocumentsContract.Document.ColumnDocumentId + $" IN ({inQuery})";
    queryArgs.PutString(ContentResolver.QueryArgSqlSelection, fileSelectionQuery);
    queryArgs.PutStringArray(ContentResolver.QueryArgSqlSelectionArgs, fileSelection);
}


  var c = AppContentResolver.Query(children, projection, queryArgs, null);```

**Comment by jpobst** [comment]
I don't think there's much we can help with on the first issue.  We simply allow the Java methods to be called from C#, so we don't really have any effect on how they behave.  This would need to be worked out with Google.

For the `Java.Lang.NoSuchMethodError` issue, this looks more like an issue in the Xamarin stuffs.  I tried it locally and it works here.  One things I noted is that method was added in API 26, so you will need to ensure your target device/emulator is running at least that version.
